services/movie.py

```diff
@@ -28,13 +33,16 @@
     genres_ids: list = None,
     actors_ids: list = None,
 ) -> Movie:
-    movie = Movie.objects.create(
-        title=movie_title,
-        description=movie_description,
-    )
-    if genres_ids:
-        movie.genres.set(genres_ids)
-    if actors_ids:
-        movie.actors.set(actors_ids)
-
-    return movie
+    with transaction.atomic():
+        movie = Movie(
+            title=movie_title,
+            description=movie_description,
+        )
+        if genres_ids:
+            movie.genres.set(genres_ids)
+        if actors_ids:
+            movie.actors.set(actors_ids)
```

Comment:
The `set` method for `genres` and `actors` should be called after the `movie` instance is saved. Call `movie.save()` before setting these relationships to ensure the instance has a primary key.